src/elements/otp-generator/otp-generator.html

```diff
@@ -45,8 +49,8 @@
       var that = this;
       new Polymer.IronMeta({type: 'otp', key: 'otp', value: this});
       this.timer = setInterval(function() {
-        var rest = 60 - (new Date()).getSeconds();
-        that.fire('timer', {remaining: rest});
+        var remaining = (60 - (new Date()).getSeconds()) % that.refreshInterval;
```

Comment:
Ohhh, I hadn't spotted that getSeconds was the seconds part of the time, not a unix timestamp.

As it is this will only work for factors of 60. Let's say that you have an interval of 40 seconds. The next code will be due at 12:00:40 and it is currently 12:00:38. So the calculation will be (60 - 38) % 40 = 22 % 40 = 22 seconds remaining, while the true answer should be 2 seconds.

Instead you want something like:
```javascript
var time = parseInt(Date.now()/1000);
var counter = Math.floor(time/that.refreshInterval);
var remaining = ((counter + 1) * that.refreshInterval) - time;
```

Also, the `getTOTP` method has a hard coded 30 too.